gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/account/AccountEndpointHandler.java

```diff
@@ -0,0 +1,151 @@
+            //todo: handle error
+        }).toSingle().subscribe(user -> {
+            routingContext.put(ConstantKeys.USER_CONTEXT_KEY, user);
+            routingContext.next();
+        });
+    }
+
+    //mounted angular
+    public void getAccount(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    //static assets
+    public void getAsset(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void updateDevices(RoutingContext routingContext) {
+        //TODO: waiting to impl fully
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        AccountResponseHandler.handleWIP(routingContext);
+    }
+
+    public void getProfile(RoutingContext routingContext) {
+        AccountResponseHandler.handleDefaultResponse(routingContext, routingContext.get(ConstantKeys.USER_CONTEXT_KEY));
+    }
+
+    public void getActivity(RoutingContext routingContext) {
+        getActivityAudit(routingContext, routingContext.get(ConstantKeys.USER_CONTEXT_KEY)).subscribe(result -> AccountResponseHandler.handleDefaultResponse(routingContext, result));
+    }
+
+    public void getDevices(RoutingContext routingContext) {
+        collectFactors(routingContext.get(ConstantKeys.USER_CONTEXT_KEY)).subscribe(factors -> AccountResponseHandler.handleDefaultResponse(routingContext, factors));
+    }
+
+    public void updateProfile(RoutingContext routingContext) {
+        User user = routingContext.get(ConstantKeys.USER_CONTEXT_KEY);
+        User updatedUser = routingContext.getBodyAsJson().mapTo(User.class);
+        if (Objects.equals(user.getId(), updatedUser.getId())) {
+            userService.update(updatedUser)
+                    .doOnSuccess(result -> AccountResponseHandler.handleUpdateUserResponse(routingContext))
+                    .doOnError(er -> AccountResponseHandler.handleUpdateUserResponse(routingContext, er.getMessage()));
+        } else {
+            AccountResponseHandler.handleUpdateUserResponse(routingContext, "Mismatched user IDs", 401);
+        }
+    }
+
+    private Single<Page<Audit>> getActivityAudit(RoutingContext routingContext, User user) {
+        int page;
+        int size;
+        try {
+            page = Integer.parseInt(routingContext.pathParam("page"));
+            size = Integer.parseInt(routingContext.pathParam("size"));
+        } catch (NumberFormatException nfe) {
+            page = DEFAULT_PAGE;
+            size = DEFAULT_PAGE_SIZE;
+        }
+
+        Single<Page<Audit>> auditPage = activityAuditService.search(
+                ReferenceType.DOMAIN,
+                domain.getReferenceId(),
+                new AuditReportableCriteria.Builder().user(user.getReferenceId()).build(),
```

Comment:
This endpoint always return an empty page, the criteria should be `user.getId()` instead of the `user.getReferenceId()`.
The referenceId represent the entity id on which the user is attached, in our case it is a domain (the referenceType attribute give this information)